### PR TITLE
New alerts API

### PR DESCRIPTION
**Pull request by alice-haber**
… new Alerts API via docs

### CODE REVIEW

File: content/api/customer-alerts.apib [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    + Health Score
+
+Alerts are configured to be account-wide for Monthly Sending Limit.
+
+## Metric Limitations
+
+For each metric group, some configuration options simply don't make sense. The following metrics have restrictions:
+
+* Real-Time
+    + interval - Must be '15 Minutes'
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: We should not expose interval in the API

File: content/api/customer-alerts.apib [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+* Real-Time
+    + Monthly Sending limit
+* Signals
+    + Health Score
+
+Alerts are configured to be account-wide for Monthly Sending Limit.
+
+## Metric Limitations
+
+For each metric group, some configuration options simply don't make sense. The following metrics have restrictions:
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: I'd phrase this the other way around: "Each metric group has a specific set of filters allowed."

File: content/api/customer-alerts.apib [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    ]
+  }
+  ```
+
+## Alert Object
+
++ Data Structure: Attributes
+    + id (number) - Unique identifier of the alert.
+    + name (string) - Display name of the alert.
+    + sid (number) - Subaccount filter applied to the alert.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: Our API convention is to write out the full word `subaccount_id`

File: content/api/customer-alerts.apib [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+    + any_sid (boolean) - Flag indicating all subaccounts should be evaluated for this alert.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: same as above `any_subaccount` (mirroring `shared_with_subaccounts`)

File: content/api/customer-alerts.apib [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+    + metric (enum) - Metric that the alert is configured to track.
+        + `"Monthly Sending Limit", "Health Score"`
+    + interval (enum) - Interval at which the alert is evaluated.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: We should not expose interval.

File: content/api/customer-alerts.apib [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+        + `"24 Hours", "15 Minutes"`
+    + evaluation_enabled (boolean) - Flag indicating whether the alert should be evaluated.
+    + filters (array) - Filter objects to apply to the alert.
+    + threshold_evaluator (object) - Evaluator object for an alert with a trigger threshold.
+      + operator (enum) - Operator for alert evaluation to use.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: API convention is lowercased values with underscores dividing words for enums

File: content/api/customer-alerts.apib [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+        + `"GT", "LT"`
+      + source (enum) - Source of value to compare against.
+        + `"Raw", "Day Over Day", "Week Over Week"`
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: API convention is lowercased values with underscores dividing words for enums

File: content/api/customer-alerts.apib [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+        though multiple values are allowed.
+
+* Monthly Sending Limit (Real-Time)
+  + filters - must be an empty array. No filters may be applied to this metric.
+
+## Filter Object
+
++ Data Structure: Attributes
+  + filter_type (enum) - Type of filter to apply to the alert.
+    + `"Sending Domain", "Mailbox Provider", "Sending IP", "IP Pool"`
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: API convention is lowercased values with underscores dividing words for enums

File: content/api/customer-alerts.apib [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+Alerts are configured to be account-wide for Monthly Sending Limit.
+
+## Metric Limitations
+
+For each metric group, some configuration options simply don't make sense. The following metrics have restrictions:
+
+* Real-Time
+    + interval - Must be '15 Minutes'
+    + filters - filter_type can be 'Sending Domain', 'Sending IP', or 'Mailbox Provider'
+    + threshold_evaluator - source must be 'Raw'
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: lowercased values with underscores dividing words

File: content/api/customer-alerts.apib [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
++ Data Structure: Attributes
+    + id (number) - Unique identifier of the alert.
+    + name (string) - Display name of the alert.
+    + sid (number) - Subaccount filter applied to the alert.
+    + any_sid (boolean) - Flag indicating all subaccounts should be evaluated for this alert.
+    + metric (enum) - Metric that the alert is configured to track.
+        + `"Monthly Sending Limit", "Health Score"`
+    + interval (enum) - Interval at which the alert is evaluated.
+        + `"24 Hours", "15 Minutes"`
+    + evaluation_enabled (boolean) - Flag indicating whether the alert should be evaluated.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: Can this parameter be simpler just named `enabled`? (No strong opinion just raising the question)

File: content/api/customer-alerts.apib [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+    + filters (array) - Filter objects to apply to the alert.
+    + threshold_evaluator (object) - Evaluator object for an alert with a trigger threshold.
+      + operator (enum) - Operator for alert evaluation to use.
+        + `"GT", "LT"`
+      + source (enum) - Source of value to compare against.
+        + `"Raw", "Day Over Day", "Week Over Week"`
+      + value (number) - Target value to compare against.
+    + email_channels (array) - Email addresses to send alert notifications to.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: Does it make sense to wrap this in a `channels` object and then in that put `emails`?

File: content/api/customer-alerts.apib [hunk 12]
```diff
@@ -0,0 +1,314 @@
+FORMAT: 1A
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: This should probably replace the current `alerts.apib` docs.

File: content/api/customer-alerts.apib [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+title: Customer-Alerts
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: We shouldn't publicly call this `Customer-Alerts`. Replace with `Alerts`

File: content/api/customer-alerts.apib [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+description: Configure real-time alerts for your SparkPost account.
+
+# Group Alerts
+Alerts can be used to get notifications regarding various activities and metrics on your SparkPost account.
+
+Alerts can be configured to send notifications via email to up to 10 email addresses, and supports the following metrics:
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: Break out "Channels" from "Metrics" as two separate headers.

File: content/api/customer-alerts.apib [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+      + `"24 Hours", "15 Minutes"`
+  + evaluation_enabled (boolean, required) - Flag indicating whether the alert should be evaluated.
+  + filters (array, required) - Filter objects to apply to the alert.
+  + threshold_evaluator (object, required) - Evaluator object for an alert with a trigger threshold.
+    + operator (enum, required) - Operator for alert evaluation to use.
+      + `"GT", "LT"`
+    + source (enum, required) - Source of value to compare against.
+      + `"Raw", "Day Over Day", "Week Over Week"`
+    + value (number, required) - Target value to compare against.
+  + email_channels (array, required) - Email addresses to send alert notifications to.
```
Comment (avigoldman) [inline content/api/customer-alerts.apib]: Should this be required for when we have future channels?

Comment (alice-haber) [reply content/api/customer-alerts.apib]: Discussed and arrived at: should be in a larger channels object which is required, and at least one channel needs to be present in the object.

File: content/api/customer-alerts.apib (hunk 1 above)
Comment (alice-haber) [reply content/api/customer-alerts.apib]: Corrected.

File: content/api/customer-alerts.apib (hunk 10 above)
Comment (alice-haber) [reply content/api/customer-alerts.apib]: Will rename to muted.

File: content/api/customer-alerts.apib (hunk 11 above)
Comment (alice-haber) [reply content/api/customer-alerts.apib]: Yes.

File: content/api/customer-alerts.apib (hunk 12 above)
Comment (alice-haber) [reply content/api/customer-alerts.apib]: Done.

File: content/api/customer-alerts.apib (hunk 13 above)
Comment (alice-haber) [reply content/api/customer-alerts.apib]: Done.

File: content/api/alerts.apib [hunk 16]
```diff
@@ -1,205 +1,322 @@
...
+* Real-Time
+    + filters - filter_type can be 'sending_domain', 'sending_ip', or 'mailbox_provider'
+    + threshold_evaluator - source must be 'raw'
+* Signals
+    + filters - filter_type can be 'sending_domain', 'ip_pool', or 'mailbox_provider'. Only one filter type can be selected,
+        though multiple values are allowed.
+
+* monthly_sending_limit
+  + filters - must be an empty array. No filters may be applied to this metric.
+  + subaccount - must be 0.
```
Comment (avigoldman) [inline content/api/alerts.apib]: subaccount `0` is different than all sending on an account.

File: content/api/alerts.apib [hunk 17]
```diff
@@ -1,205 +1,322 @@
...
-    + email_addresses (array) - List of email addresses to send alert notifications.
-    + alert_metric (enum) - Metric that the alert is configured to track.
-        + `"monthly_sending_limit"`
-    + threshold (object) - Configuration for level and threshold at which an alert is generated.
-        + error (object) - Error level alert configuration.
-            + target (number) - Percentage value at which an alert will be generated.
-            + comparator (enum) - Conditional comparison operator used by alerts service to compare actual value of `alert_metric` to the `target` value.
-                + `"gt"`
-    + enabled (boolean) - Whether or not the alert is active.
+    + subaccount (number) - Subaccount filter applied to the alert.
```
Comment (avigoldman) [inline content/api/alerts.apib]: Is subaccount different than a regular filter?

File: content/api/alerts.apib [hunk 18]
```diff
@@ -1,205 +1,322 @@
...
+            "test@sparkpost.com",
+            "bigwig@test.com"
+        ]
     }
-    ```
+  }
+  ```
 
-### Create an Alert [POST /labs/alerts]
+### Create an Alert [POST /v1/customer-alerts]
```
Comment (avigoldman) [inline content/api/alerts.apib]: Should be `/v1/alerts` since the old one is still under `/labs/alerts`

File: content/api/alerts.apib (hunk 17 above)
Comment (alice-haber) [reply content/api/alerts.apib]: Yes, in that both subaccount and up to 1 filter may be set for Signals alerts.

File: content/api/alerts.apib (hunk 18 above)
Comment (alice-haber) [reply content/api/alerts.apib]: Yeah, good catch

File: content/api/alerts.apib (hunk 16 above)
Comment (alice-haber) [reply content/api/alerts.apib]: Agreed on a new phrasing/structure

File: content/api/alerts.apib [hunk 19]
```diff
@@ -1,205 +1,497 @@
...
-                + Default: `"gt"`
-    + enabled (boolean) - Whether or not the alert is active.
-        + Default: true
+        + `"block_bounce_rate"`
+        + `"hard_bounce_rate"`
+        + `"soft_bounce_rate"`
+        + `"health_score"`
+    + muted (boolean, required) - Flag indicating whether the alert should be ignored.
+    + filters (array, required) - Filter objects to apply to the alert.
+    + threshold_evaluator (object, required) - Evaluator object for an alert with a trigger threshold.
```
[the remark this answers is not in the document]
Comment (jgzamora) [reply content/api/alerts.apib]: Oh I think I misread that, it's for alerts that have thresholds, that makes sense but still might rephrase

**Pull request closed without merging**